### PR TITLE
Create needs and needs/goods form

### PR DESCRIPTION
**Pull request by nstjean**
I've created the Need-Initiate form at /need/
FireShot Capture 292 - Nonprofit Exchange Hub - localhost

And then the Need-Goods form at /need/goods/
FireShot Capture 294 - Nonprofit Exchange Hub - localhost

I've tried to keep all the logic for the form in NeedFormGoods, with custom components for the TextField, Select, Radio and FileUpload. Hopefully now I can make the other forms pretty quickly. :)

Later I'm going to try to pull the styling out of NeedFormGoods into the theme, if possible.

### CODE REVIEW

File: client/src/NeedFormGoods.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    needType: '',
+    deliveryMethod: '',
+};
+
+function NeedForm() {
+    const classes = useStyles();
+
+    const [ formData, setFormData ] = React.useState(initialFormData);
+
+    const handleChange = (evt: React.ChangeEvent<HTMLInputElement>): void => {
```
Comment (jd2rogers2) [inline client/src/NeedFormGoods.tsx:109]: got TS here on compile
for the typing of this function and where it's used down the pipeline
let's just follow MUI docs https://v4.mui.com/components/selects/ for now (despite the `unknown`)
(see one of the snippets and then select TS)

File: client/src/Helpers/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default as setDefaults } from './setDefault';
```
Comment (jd2rogers2) [inline client/src/Helpers/index.ts:1]: i think this whole file and folder need to be deleted

File: client/src/NeedFormGoods.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        }));
+    };
+
+    return (
+        <>
+            <Container className={classes.formBox}>
+                <Typography variant="h2" component="h2" align="left">Share a Need: Goods</Typography>
+                <Box className={classes.borderBox}>
+                <form className={classes.root}>
+                    <Grid container spacing={5} className={classes.grid}>
```
Comment (jd2rogers2) [inline client/src/NeedFormGoods.tsx:123]: error on compile here too, no `grid` key in the classes obj

**Review by jd2rogers2 — CHANGES_REQUESTED** [review]

File: client/src/FormElements/CustomSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import { Select, OutlinedInput, MenuItem, FormControl, FormLabel } from '@material-ui/core';
+
+type CustomProps = {
+    id: string,
+    label: string,
+    placeholder: string,
+    options: {
+        value: string,
+        text: string
+    }[],
```
[the remark this answers is not in the document]
Comment (nstjean) [reply client/src/FormElements/CustomSelect.tsx:11]: Sounds fun. I'll try it out!

File: client/src/NeedFormGoods.tsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+                            <CustomFileUpload
+                                label="Photos"
+                                id="photos"
+                                text="Click here to upload photos"
+                                onChange={handleChange}
+                            />
+                        </Grid>
+                        <Grid item container xs={12} className={classes.submitbuttons}>
+                            <Grid item>
+                                <Button variant='contained' color='secondary'>
```
[the remark this answers is not in the document]
Comment (jd2rogers2) [reply client/src/NeedFormGoods.tsx]: haha i guess so, not sure really. i just know that everyone does it

File: client/src/FormElements/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    placeholder: string,
+    options: {
+        value: string,
+        text: string
+    }[],
+    value: string,
+    onChange: (event: any) => void,
+};
+
+const handleChange = (event: React.ChangeEvent<{ name?: string | undefined; value: unknown; }>): void => {
```
Comment (jd2rogers2) [inline client/src/FormElements/Select.tsx:17]: not being used, remove

**Review by jd2rogers2 — APPROVED** [review]
one more piece of clean up then 🚀

**Pull request merged**